cmd/decode.go

```diff
@@ -95,21 +104,71 @@ func validateAndRun() error {
 	return run(options.metaOnly, outMediaType, in, os.Stdout)
 }
 
+// Original requirement from etcd WAL log analysis tool etcd-dump-logs,
+// to serve as a decoder to decode k8s objects from each entry of the etcd WAL log.
+// Read data from os.Stdin, decode object line by line, and print out in os.Stdout
+func runInBatchMode(metaOnly bool, outMediaType string, out io.Writer) error {
+	var err error
+	batchmode := true
+	inputReader := bufio.NewReader(os.Stdin)
+	for {
+		input, err := inputReader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			fmt.Printf("EOF\n")
+			return nil
+		}
+		if err != nil {
+			fmt.Printf("Read Error: %s\n", err)
+			return err
+		}
+
+		input = stripNewline(input)
+		decodedinput, err := hex.DecodeString(string(input))
+		if err != nil {
+			fmt.Println(err)
```

Comment:
Let's not print errors that we return. Our top level error handler should capture and print them.  To add context to errors, wrap them, e.g. `return fmt.Errorf("error decoding input on line %d of --batch-process input: %v", ...)`